c/extras/aggregator.cc

```diff
@@ -24,35 +24,13 @@ PyObject* aggregate(PyObject*, PyObject* args) {
                         &dt, &n_bins, &nx_bins, &ny_bins, &max_dimensions, &seed)) return nullptr;
 
   DataTable* dt_in = PyObj(dt).as_datatable();
-  DataTable* dt_exemplars = nullptr;
-  DataTable* dt_members = nullptr;
-  Column** cols_exemplars = dt::amalloc<Column*>(dt_in->ncols + 1);
-  Column** cols_members = dt::amalloc<Column*>(static_cast<int64_t>(2));
+  DataTablePtr dt_members = nullptr;
 
-  for (int32_t i = 0; i < dt_in->ncols; ++i) {
-    LType ltype = info(dt_in->columns[i]->stype()).ltype();
-    switch (ltype) {
-      case LType::BOOL:
-      case LType::INT:
-      case LType::REAL: cols_exemplars[i] = dt_in->columns[i]->cast(SType::FLOAT64); break;
-      default:          cols_exemplars[i] = dt_in->columns[i]->shallowcopy();
-    }
-  }
-
-  cols_exemplars[dt_in->ncols] = nullptr;
-  cols_members[0] = Column::new_data_column(SType::INT32, dt_in->nrows);
-  cols_members[1] = nullptr;
-  dt_exemplars = new DataTable(cols_exemplars);
-  dt_members = new DataTable(cols_members);
 
   Aggregator agg(n_bins, nx_bins, ny_bins, max_dimensions, seed);
-  agg.aggregate(dt_exemplars, dt_members);
+  dt_members = DataTablePtr(agg.aggregate(dt_in));
 
-  PyObject *res = PyTuple_New(2);
-  PyTuple_SET_ITEM(res, 0, pydatatable::wrap(dt_exemplars));
-  PyTuple_SET_ITEM(res, 1, pydatatable::wrap(dt_members));
-
-  return res;
+  return pydatatable::wrap(dt_members.release());
```

Comment:
Not sure what's the point in creating an `std::unique_ptr<>` only to release in the very next line.

Reply:
Yes, that's just an artifact. Fixed.

Reply:
`dt_members` still remains a `std::unique_ptr`, and you're still calling `.release()` on it...